src/batHeap.ml

```diff
@@ -107,34 +107,40 @@ let find_min bh = match bh.mind with
 *)
 
 
-let rec find_min_tree ts k = match ts with
-  | [] -> failwith "find_min_tree"
+let rec find_min_tree fail ts k = match ts with
```

Comment:
I'm sorry for the continued nitpicking, but the two continuation arguments should really follow each other, currently the ordering makes no sense.

In general I would maybe expect the success continuation first and then the failure continuation, but given the current calling style (which is not unrelated to a monadic let),  think that having the failure continuation first makes sense (that is: makes call sites look nicer than with the other choice). It also corresponds to the ordering `None | Some of 'a`, which is not unnatural.

Also, I think it would be nice to name it `kfail` so that people recognize the two-continuations pattern.

Reply:
> I'm sorry for the continued nitpicking, but the two continuation arguments should really follow each other, currently the ordering makes no sense.

Whoops: I was too rushed, and didn't notice that the style is clumsy. Please, see my last changes; and thanks for the nitpicking -- details matter.